### DISCUSSION TITLE
[dove] how to distinguish service caller

### DISCUSSION THREAD

**Discussion by anonimusprogramus** [feathersjs/feathers#2930] [Q&A]
(Using socketio transport) how does a service knows if its method is being called **from feathers socketio-client** or internally in server **from another services** ?

I know (optionally) I can give a `params.query` from client to distinguish, but in this case no query, let's say just simple `.find() `

Thanks

**Comment by Mairu** [comment]
You can check for params.provider -> https://docs.feathersjs.com/api/express.html#params-provider

**Reply by anonimusprogramus** [reply]
thanks, I have checked it, but both (client & internal) still output the same, `{ provider: 'socketio' } `

**Reply by Mairu** [reply]
Mhh, should not be the case, only if you would copy over the params and provide them to the internal service call yourself.
